server-spi/src/main/java/org/keycloak/sessions/AuthenticationSessionProvider.java

```diff
@@ -31,27 +31,87 @@
     /**
      * Creates and registers a new authentication session with random ID. Authentication session
      * entity will be prefilled with current timestamp, the given realm and client.
+     * @param realm {@code RealmModel}
+     * @return Returns created {@code RootAuthenticationSessionModel}. Never returns {@code null}.
      */
     RootAuthenticationSessionModel createRootAuthenticationSession(RealmModel realm);
 
-    RootAuthenticationSessionModel createRootAuthenticationSession(String id, RealmModel realm);
+    /**
+     * Creates a new root authentication session specified by the provided id and realm.
+     * @param id {@code String}
+     * @param realm {@code RealmModel}
+     * @return Returns created {@code RootAuthenticationSessionModel}. Never returns {@code null}.
+     * @deprecated Use {@link #createRootAuthenticationSession(RealmModel, String)} createRootAuthenticationSession} instead.
+     */
+    @Deprecated
+    default RootAuthenticationSessionModel createRootAuthenticationSession(String id, RealmModel realm) {
+        return createRootAuthenticationSession(realm, id);
+    }
 
+    /**
+     * Creates a new root authentication session specified by the provided realm and id.
+     * @param realm {@code RealmModel}
+     * @param id {@code String}
+     * @return Returns created {@code RootAuthenticationSessionModel}. Never returns {@code null}.
+     */
+    RootAuthenticationSessionModel createRootAuthenticationSession(RealmModel realm, String id);
+
+    /**
+     * Returns the root authentication session specified by the provided realm and id.
+     * @param realm {@code RealmModel}
+     * @param authenticationSessionId {@code RootAuthenticationSessionModel}
+     * @return Returns found {@code RootAuthenticationSessionModel} or {@code null} if no root authentication session is found.
+     */
     RootAuthenticationSessionModel getRootAuthenticationSession(RealmModel realm, String authenticationSessionId);
 
-    void removeRootAuthenticationSession(RealmModel realm, RootAuthenticationSessionModel authenticationSession);
+    /**
+     * Removes provided root authentication session.
+     * @param realm {@code RealmModel}
+     * @param authenticationSession {@code RootAuthenticationSessionModel}
+     * @deprecated Use {@link #removeRootAuthenticationSession(RootAuthenticationSessionModel)} removeRootAuthenticationSession} instead.
+     */
+    @Deprecated
+    default void removeRootAuthenticationSession(RealmModel realm, RootAuthenticationSessionModel authenticationSession) {
+        removeRootAuthenticationSession(authenticationSession);
+    }
+
+    /**
+     * Removes provided root authentication session.
+     * @param authenticationSession {@code RootAuthenticationSessionModel}
+     */
+    void removeRootAuthenticationSession(RootAuthenticationSessionModel authenticationSession);
```

Comment:
Why has this method been introduced?

Reply:
Because I deprecated `removeRootAuthenticationSession(RealmModel realm, RootAuthenticationSessionModel authenticationSession)` as realm parameter is not used.

Reply:
Realm parameter is not used in the current implementation but is necessary to be retained for larger deployments with different logic of distributing sessions could leverage it.